src/routes/auth/data.ts

```diff
@@ -0,0 +1,10 @@
+import type { SignInData } from './validate.ts';
+
+export async function authenticate(database: D1Database, body: SignInData) {
+	const { results } = await database
+		.prepare('SELECT id FROM profile WHERE email = ? AND password = ?')
+		.bind(body.email, body.password)
```

Comment:
consider hashing passwords with bcrypt or argon2!

Reply:
We decided to use PBKDF2 as it's the only native solution suitable for hashing passwords stored in a database. Other alternatives would require Rust-based implementations, which we couldn't get working.